test/node-rpc-test.js

```diff
@@ -222,6 +224,21 @@ describe('RPC', function() {
       connectionsCnt = await nclient.execute('getconnectioncount', []);
       assert.strictEqual(connectionsCnt, 0);
     });
+
+    it('should rpc getnodeaddresses', async () => {
+      const newHost = {host:'223.190.4.107', port:48444, time:1656165403, services: 1149};
+      const netaddress = new NetAddress(newHost);
+      const isHostAdded = node.pool.hosts.add(netaddress);
+
+      assert.strictEqual(isHostAdded, true);
+
+      const hosts = await nclient.execute('getnodeaddresses');
```

Comment:
It would be great to see a second test that makes use of the "count" parameter as well. Maybe you can use a loop to add like 10 random addrs to the hostlist, then execute this api call with arguments 0, 1, 10, 20 or something?